canonical/cos-alerter
# Adding PagerDuty Native support

## Issue
Fixes #61 

Raised pagerduty alerts are not using dedupication key to correlating triggers and resolves.
Causing a new alerts to be trigered every interval and alerts are never resolved


## Solution
Native PagerDuty support is added to cos-alerter via the PD python client [pdpyras](https://github.com/PagerDuty/pdpyras)

## Context
Currently we're depending on Apprise to send notifications to destinations for when an alertmanager instance is down, Apprise currently has no support for adding dedup key to the PagerDuty incident


## Testing Instructions
1- Deploy cos-alerter and add a Pagerduty valid connection string `pagerduty://{integration_key}@{api_key}` to cos-alerter list of destinations 
2- PagerDuty incidents should be grouped per an Alertmanager client if it's down
3- Once the Alertmanager client is up again the PD incident should be self-resolved
